conanfile.py

```diff
@@ -119,10 +120,30 @@ class BasiliskConan(ConanFile):
         pass
 
     def system_requirements(self):
+        if not self.options.managePipEnvironment:
+            return  # Don't need to manage any pip requirements
+
+        # TODO: Remove everything here, which only runs if we have set
+        # managePipEnvironment (i.e. conanfile.py-based build).
+
+        # ensure latest pip is installed
+        if is_running_virtual_env() or platform.system() == "Windows":
+            cmakeCmdString = 'python -m pip install --upgrade pip'
+        else:
+            cmakeCmdString = 'python3 -m pip install --upgrade pip'
```

Comment:
I think it might be good to make the `--managePipEnvironment` an opt-in flag rather than the default, so that the user must explicitly give the conanfile permission to modify the `pip` installation directly. But if the current way has been working well for developers until now, maybe it's fine to leave it as is. The recommended practice is for users to have the latest `pip` at all times anyway, so maybe it's fine to continue to do it like this.

Once we manage to have binary wheels built and uploaded to Pypi, most users will be able to just `pip install Basilisk` directly without ever even seeing this conanfile, so these things are mainly just for power users and developers anyway.

Reply:
Ok, let's leave it as is then for now.